tsl/src/remote/dist_ddl.c

```diff
@@ -603,21 +601,71 @@ dist_ddl_process_grant_on_table(ProcessUtilityArgs *args)
 }
 
 static void
-dist_ddl_process_grant(ProcessUtilityArgs *args)
+dist_ddl_process_grant_object(const ProcessUtilityArgs *args)
 {
-	GrantStmt *stmt = castNode(GrantStmt, args->parsetree);
+	const GrantStmt *stmt = castNode(GrantStmt, args->parsetree);
 
-	if (stmt->objtype == OBJECT_DATABASE)
-		dist_ddl_process_grant_on_database(args);
-	else if (stmt->objtype == OBJECT_TABLE)
+	switch (stmt->objtype)
 	{
-		if (stmt->targtype == ACL_TARGET_ALL_IN_SCHEMA)
-			dist_ddl_process_grant_on_schema(args);
-		else
+		case OBJECT_DATABASE:
+			dist_ddl_process_grant_on_database(stmt);
+			break;
+		case OBJECT_TABLE:
 			dist_ddl_process_grant_on_table(args);
+			break;
+		default:
+			break;
+	}
+}
+
+static void
+dist_ddl_process_grant_all_in_schema(const ProcessUtilityArgs *args)
+{
+	const GrantStmt *stmt = castNode(GrantStmt, args->parsetree);
+
+	switch (stmt->objtype)
+	{
+		case OBJECT_TABLE:
+			dist_ddl_process_grant_on_schema(args);
```

Comment:
Just to note that the naming of this function is actually not correct, I've fixed it in my PR. This is still grant on table command but with all in schema clause. `GRANT ON SCHEMA` is a different command

Reply:
Ok, I'll let you fix it then...